queries/haskell/highlights.scm

```diff
@@ -186,59 +239,67 @@
 ((signature (variable) @_type (forall (context (fun)))) . (function (variable) @function) (#eq? @function @_type))
 
 ; consider infix functions as operators
-(exp_infix (variable) @operator)  
-(exp_infix (qualified_variable (variable) @operator))
+(exp_infix [
+  (variable) @operator
+  (qualified_variable (variable) @operator)
+])
 ; partially applied infix functions (sections) also get highlighted as operators
-(exp_section_right (variable) @operator) 
-(exp_section_right (qualified_variable (variable) @operator)) 
-(exp_section_left (variable) @operator)
-(exp_section_left (qualified_variable (variable) @operator))
+(exp_section_right [
+  ((variable) @operator)
+  (qualified_variable (variable) @operator)
+  ]) 
+(exp_section_left [
+  ((variable) @operator)
+  (qualified_variable (variable) @operator)
+  ])
 
 ; function calls with an infix operator
 ; e.g. func <$> a <*> b
-(exp_infix (exp_name) @function.call . (operator))
-; qualified function calls with an infix operator
-(exp_infix (exp_name 
-  (qualified_variable (
+(exp_infix (exp_name [
+    ((variable) @function.call)
+    (qualified_variable (
       (module) @namespace
-      (variable) @function.call))) . (operator))
+      (variable) @function.call))
+  ]) . (operator))
```

Comment:
This may be easier to keep track of which part of the query is for which

```suggestion
(exp_infix
  (exp_name
    [
      ((variable) @function.call))
      (qualified_variable
        (module) @namespace
        (variable) @function.call)
    ])
  . (operator))
```